snuba/datasets/cdc/message_filters.py

```diff
@@ -0,0 +1,39 @@
+from snuba import environment
+from snuba.datasets.message_filters import StreamMessageFilter
+from snuba.utils.metrics.backends.wrapper import MetricsWrapper
+from snuba.utils.streams.kafka import KafkaPayload
+from snuba.utils.streams.types import Message
+
+metrics = MetricsWrapper(environment.metrics, "cdc.consumer")
+
+KAFKA_ONLY_PARTITION = (
+    0  # CDC only works with single partition topics. So partition must be 0
+)
+
+
+class CdcTableNameMessageFilter(StreamMessageFilter[KafkaPayload]):
+    """
+    Filters CDC messages not related with the table this filter is interested into.
+    This filtering happens based on the table header when present and without
+    parsing the message payload.
+    """
+
+    def __init__(self, postgres_table: str) -> None:
+        self.__postgres_table = postgres_table
+
+    def should_drop(self, message: Message[KafkaPayload]) -> bool:
+        assert (
+            message.partition.index == KAFKA_ONLY_PARTITION
+        ), "CDC can only work with single partition topics for consistency"
+
+        table_header = next(
+            (header for header in message.payload.headers if header[0] == "table"), None
+        )
+
+        if table_header:
+            table_name = table_header[1].decode("utf-8")
+            if table_name != self.__postgres_table:
+                metrics.increment("CDC Message Dropped", tags={"table": table_name})
```

Comment:
Does datadog actually allow string with spaces? I've never looked, but it seems to go against our typical usage pattern.

Reply:
good point. statsd did, datadog may not.